pyxrf/model/command_tools.py

```diff
@@ -208,40 +223,192 @@ def pyxrf_batch(start_id, end_id=None, wd=None, fit_channel_sum=True, param_file
+
+        if start_id is None and end_id is None:
+            # ``start_id`` and ``end_id`` are not specified:
+            #   process all .h5 files in the current directory
+            flist = all_files
+        elif end_id is None:
+            # only ``start_id`` is specified:
+            #   process only one file that contains ``start_id`` in its name
+            #   (only if such file exists)
+            flist = [fname for fname in all_files
+                     if re.search(f"^[^_]*_{str(start_id)}\D+", fname)]  # noqa: W605
+            if len(flist) < 1:
+                print(f"File with Scan ID {start_id} was not found.")
+            else:
+                print(f"Processing file with Scan ID {start_id}")
+        else:
+            # ``start_id`` and ``end_id`` are specified:
+            #   select files, which contain the respective ID substring in their names
+            flist = []
+            for data_id in range(start_id, end_id+1):
+                flist += [fname for fname in all_files
+                          if re.search(f"^[^_]*_{str(data_id)}\D+", fname)]  # noqa: W605
+            if len(flist) < 1:
+                print(f"No files with Scan IDs in the range {start_id} .. {end_id} were found.")
+            else:
+                print(f"Processing file with Scan IDs in the range {start_id} .. {end_id}")
+
+    # Check if .json parameter file exists
+    if not os.path.isabs(param_file_name):
+        pname = os.path.join(wd, param_file_name)
+    else:
+        pname = param_file_name
+    if not os.path.exists(pname) or not os.path.isfile(pname):
+        print(f"ERROR: can't find the parameter file {pname}. "
+              "Check the value of 'param_file_name'.")
+        raise IOError(f"Function 'pyxrf_batch'. Processing parameter file '{pname}' does not exist.")
+    else:
+        print(f"Processing parameter file: {pname}")
+
+    if len(flist) > 0:
+
+        print(f"The following files are scheduled for processing:")
+        for fln in flist:
+            print(f"    {fln}")
+        print(f"Total number of selected files: {len(flist)}")
+        print()
+
         for fpath in flist:
+            print(f"Processing file '{fpath}' ...")
             fname = fpath.split(sep_v)[-1]
             working_directory = fpath[:-len(fname)]
             fit_pixel_data_and_save(working_directory, fname,
                                     fit_channel_sum=fit_channel_sum, param_file_name=param_file_name,
                                     fit_channel_each=fit_channel_each, param_channel_list=param_channel_list,
                                     incident_energy=incident_energy, spectrum_cut=spectrum_cut,
                                     save_txt=save_txt, save_tiff=save_tiff,
-                                    ic_name=ic_name)
+                                    ic_name=ic_name, use_average=use_average)
+
+        print()
+        print("All selected files were processed.")
```

Comment:
Maybe better to use one print with `\n`?

Reply:
I am not sure. I always used to think, that empty prints are easier to find, insert, delete and comment out in the code, especially if there is a long sequence of prints. Empty prints show the locations of empty lines in the output.